src/cgame/cg_rocket_progressbar.cpp

```diff
@@ -89,15 +89,27 @@ static float CG_Rocket_GetStaminaProgress()
 
 static float CG_Rocket_GetPlayerOxygenProgress()
 {
+	int outTime = cg.snap->ps.lowOxygenTime;
+
 	// regular progress bar status
-	if ( cg.underwater )
+	if ( outTime )
 	{
-		float underwaterDuration = cg.time - cg.underwaterTime;
-		return 1.0f - underwaterDuration / static_cast<float>( OXYGEN_MAX_TIME );
+		// the time number is sent cropped, we have to do modulo arithmetics to determine the delta
+		int timeLeft = ( outTime - cg.time ) & ((1 << LOW_OXYGEN_TIME_BITS) - 1);
+
+		if ( timeLeft > OXYGEN_MAX_TIME )
+		{
+			// this happens when we just took damage but we haven't
+			// received the new time yet. Let's do some prediction.
+			timeLeft = (timeLeft + 1000) & ((1 << LOW_OXYGEN_TIME_BITS) - 1);
```

Comment:
If the player is already drowning you should just set the time left to 0. And also make 1000 the zero point of time left since you can't tell whether you're really drowning already or not when it's under 1000. It does not make sense for the progressbar to go back and forth a little while already drowning

Reply:
Well I like to know when I will take damage next when I'm in game, and this works out nicely for that IMO

Reply:
Also note that what you are suggesting would work only for a few frames, until we receive the new oxygen reserve value from the server — and here we will have to show the bar non-empty

Reply:
I mean something like:
```
int timeLeft = ( outTime - cg.time ) & ((1 << LOW_OXYGEN_TIME_BITS) - 1);
if (timeLeft > OXYGEN_MAX_TIME) //underflow
    timeLeft = 0;
else
    timeLeft = std::max(0, timeLeft - 1000);
```

As a user it would seem simply wrong and buggy to me if the bar is going back and forth while drowning.